STRATUM.md

```diff
@@ -0,0 +1,368 @@
+- [client.reconnect](#client-reconnect)
+
+
+### Errors
+
+Whenever an RPC call triggers an error, the response MUST include an `error`
+field which maps to a **list** of the following values:
+
+- [ `code` : `int` ]
+- [ `message` : `string` ]
+- [ `data` : `object` ]
+
+```
+{"id": 10, "result": null, "error": [21, "Job not found", null]}
+```
+
+Errors SHOULD be identified by their `code` and programs SHOULD do error
+handling based on the `code` and not the `message`.
+Available error codes, in addition to the codes defined in the [JSON RPC 2.0]() specification, are:
+
+- `20` - Other/Unknown
+- `21` - Job not found (=stale)
+- `22` - Duplicate share
+- `23` - Low difficulty share
+- `24` - Unauthorized worker
+- `25` - Not subscribed
+
+The `message` field SHOULD be a concise description of the error for human
+consumption.
+
+Implementors MAY choose to include an optional `data` object with additional
+information relevant to the error.
+
+
+### mining.configure
+
+In order to allow easier upgrades in the future, a client MAY send a
+configuration as its initial messages to the server, specifying possible
+extensions it supports.
+The specific mechanism is described by [slushpool here](https://github.com/slushpool/stratumprotocol/blob/master/stratum-extensions.mediawiki)
+
+A client MAY choose to skip this message, which a server SHOULD interpret as the
+client adhering to the protocol as described in this document, without any
+further extensions.
+
+This method call will only be executed by clients.
+
+
+### mining.subscribe
+
+In order to initiate or resume a session with the server, a client needs to
+call the subscribe method.
+
+This method call will only be executed by clients.
+
+
+#### Request:
+
+```
+{"id": 1, "method": "mining.subscribe", "params": ["MyMiner/1.0.0", null, "my.pool.com", 1234]}\n
+```
+
+
+- [ `id` : `int` ]: request id
+- [ `method` : `string` ]: RPC method name
+- [ `params` : (`string`, `string`, `string`, `int`) ]: list of method
+  parameters
+	1. MUST be name and version of mining software in the given format or empty
+	   string
+	2. MUST be session id received during a previous session or `null` if a new
+	   session should be initiated by the server
+	3. MUST be the host the client is trying to connect to or `null`
+	4. MUST be the port on the given host the client is trying to connect to or
+     `null`
+
+Please see the [zcash stratum specification](https://github.com/str4d/zips/blob/77-zip-stratum/drafts/str4d-stratum/draft1.rst#rationale)
+for a reason as to why `subscribe` includes a host and port.
+
+
+#### Response
+
+```
+{ "id": 1, "result": ["ed70df83","080000"], "error": null }\n
+```
+
+- [ `id` : `int` ]: request id
+- [ `result` : (`string`, `string`) ]:
+	- MUST be `null` if an error occurred or otherwise
+		1. If the server supports session resumption, then this SHOULD be a unique
+       session id, `null` otherwise
+    2. The first bytes of the 8 byte nonce, hex encoded
```

Comment:
A markup fix (for readability).
```suggestion
		2. The first bytes of the 8 byte nonce, hex encoded
```

Reply:
Clarification re order of bytes (and markup fix). A couple of lines below, this doc says that this value "is usually referred to as the extranonce" and notes `nonce = minernonce || extranonce` so seems in contrast with this line. I suggest:
```suggestion
		2. The last bytes of the 8 byte nonce, hex encoded
```

@skkw @sennui I considered:
* [Current dev stratum PR](https://github.com/aeternity/aeternity/pull/2531/files#diff-3f2e90c96b318c0fd0c2dcd5c04a4003R742).
* [Lib used in that PR](https://github.com/aeternity/aestratum_lib/blob/96ffd9a36fab0d65640e87000e8466458276b8f3/src/aestratum_nonce.erl#L198-L207).